packages/server/lib/controllers/proxy.controller.ts

```diff
@@ -0,0 +1,320 @@
+import type { Request, Response } from 'express';
+import type { OutgoingHttpHeaders } from 'http';
+import type { NextFunction } from 'express';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import axiosRetry from 'axios-retry';
+
+import logger from '../utils/logger.js';
+import errorManager from '../utils/error.manager.js';
+import configService from '../services/config.service.js';
+import { getAccount } from '../utils/utils.js';
+import type { ProxyBodyConfiguration, Connection, HTTP_VERB } from '../models.js';
+import { NangoError } from '../utils/error.js';
+import { getConnectionCredentials } from '../utils/connection.js';
+
+const RETRIES = 2;
+
+axiosRetry(axios, {
+    retries: RETRIES,
+    retryDelay: axiosRetry.exponentialDelay,
+    retryCondition: (error) => {
+        const status = error?.response?.status;
+        if (error?.response?.status.toString().startsWith('5') || error?.response?.status === 429) {
+            logger.info(`API received a ${status} error, retrying with exponential backoffs for a total of ${RETRIES} times`);
+            return true;
+        }
+        return false;
+    }
+});
+
+class ProxyController {
+    public async routeSDKCall(req: Request, res: Response, next: NextFunction) {
+        if (req.body === null) {
+            errorManager.errRes(res, 'missing_body');
+            logger.error(`Proxy: a POST body with the needed configuration is missing.`);
+            return;
+        }
+        const { valid, error } = this.validateBody(req);
+        if (!valid) {
+            errorManager.errRes(res, String(error));
+            return;
+        }
+
+        const configBody = { ...req.body };
+        const { method } = configBody;
+
+        logger.info(`Proxy: received ${method} call with a valid body`);
+
+        const { providerConfigKey } = configBody;
+        const accountId = getAccount(res);
+        const providerConfig = await configService.getProviderConfig(providerConfigKey, accountId);
+        if (!providerConfig) {
+            logger.error(`Proxy: provider configuration not found`);
+            res.status(404).send();
+        }
+        const template = configService.getTemplate(String(providerConfig?.provider));
+
+        if (!template.base_api_url) {
+            logger.error(`Proxy: base api url configuration is missing. The providers.yaml might be missing a value`);
```

Comment:
Realistically, there will be a bunch of APIs for which we don't have a `base_api_url` for the time it takes to backfill. 

Maybe it's worth it's worth expliciting this by saying something like:  "The proxy is not supported for this provider. You can easily add support by following the instructions at https://docs.nango.dev/contribute-api." 

And we could update the ["Contribute a new API Prodider"](https://docs.nango.dev/contribute-api) docs page to have a section on how to support the proxy for an API (basically adding the base URL, in case the request signing is standard, i.e. Bearer token).

What do you think?

Reply:
🙏